### PR TITLE
Improved error message in component_mask.h

### PR DESCRIPTION
**Pull request by rrgrove6**

### CODE REVIEW

File: include/deal.II/fe/component_mask.h [hunk 1]
```diff
@@ -204,7 +204,8 @@ class ComponentMask
   /**
    * Exception
    */
-  DeclException0 (ExcNoComponentSelected);
+  DeclExceptionMsg (ExcNoComponentSelected, "The number of selected
+                    components in a mask must be greater than zero.");
```
Comment (bangerth) [inline include/deal.II/fe/component_mask.h]: You use a multiline string `"...\n..."`. This violates the C++ standard, although it is accepted by some compilers. The way you should write this is
```
  DeclExceptionMsg (Exc...,
                                  "abc "
                                  "def");
```
C++ says that multiple strings enclosed in double quotes will simply be collated into one, which is here `"abc def"`. This also explains why I have a space at the end of the first string: I simply want this to be a space between individual words.

**Comment by tamiko** [comment]
@rrgrove6 Can we please stick to sensible pull request summaries, i.e., remove the "because I want dinner".

**Comment by guidokanschat** [comment]
I agree with @tamiko. This is a professional repository and a few hundred people world wide depend on its functioning. It is **not** intended for practicing pull requests or similar!

**Comment by rrgrove6** [comment]
I believe I fixed the problem @bangerth.

@tamiko and @guidokanschat, it was just a joke.. I planned on changing it before we merged it..

**Pull request merged**